services/appveyor/appveyor-build-job.service.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+const { renderBuildStatusBadge } = require('../build-status')
+const AppVeyorBase = require('./appveyor-base')
+const { NotFound } = require('..')
+
+module.exports = class AppVeyorBuildJob extends AppVeyorBase {
+  static get route() {
+    return {
+      base: 'appveyor/job/build',
+      pattern: ':user/:repo/:job/:branch*',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'AppVeyor Job',
+        pattern: ':user/:repo/:job',
+        namedParams: {
+          user: 'wpmgprostotema',
+          repo: 'voicetranscoder',
+          job: 'Linux',
+        },
+        staticPreview: renderBuildStatusBadge({ status: 'success' }),
+      },
+      {
+        title: 'AppVeyor Job branch',
+        pattern: ':user/:repo/:job/:branch',
+        namedParams: {
+          user: 'wpmgprostotema',
+          repo: 'voicetranscoder',
+          job: 'Windows',
+          branch: 'master',
+        },
+        staticPreview: renderBuildStatusBadge({ status: 'success' }),
+      },
+    ]
+  }
+
+  transform({ data, jobName }) {
+    if (!data.hasOwnProperty('build')) {
+      // this project exists but no builds have been run on it yet
+      return { status: 'no builds found' }
+    }
+
+    const {
+      build: { jobs },
+    } = data
+    if (!jobs) {
+      throw new NotFound({ prettyMessage: 'no jobs found' })
+    }
```

Comment:
TBH I'm skeptical there's a real scenario where we'd ever see this, but given the array of TypeError's we see in prod I figured better safe than sorry here

Reply:
That was my initial thought as well, and if we were rolling out the AppVeyor support for the very first time I would've wanted to structure the schema that way.

I decided not to do that here though because the existing AppVeyor badges that have been in use do not enforce this at the schema level, and the [API documentation](https://www.appveyor.com/docs/api/projects-builds/#get-project-last-build) I could find does not provide much details around the response schema (just one example response). 

Given that uncertainty, my thinking is that it'd be better to handle this case programmatically instead of potentially causing `invalid response data` badges, especially on the existing AppVeyor Build/Test badges.